dashboard/src/app/workspaces/workspace-details/environments/environments.controller.js

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2015-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+/**
+ * @ngdoc controller
+ * @name workspace.details.controller:WorkspaceEnvironmentsController
+ * @description This class is handling the controller for details of workspace : section environments
+ * @author Oleksii Kurinnyi
+ */
+export class WorkspaceEnvironmentsController {
+
+  /**
+   * Default constructor that is using resource injection
+   * @ngInject for Dependency injection
+   */
+  constructor($q, $log, $scope, $timeout) {
+    this.$q = $q;
+    this.$log = $log;
+    this.$scope = $scope;
+
+    this.editorOptions = {
+      lineWrapping: true,
+      lineNumbers: false,
+      mode: 'text/x-yaml',
+      readOnly: true,
+      onLoad: (editor) => {
+        $timeout(() => {
+          editor.refresh();
+        }, 1000);
+      }
+    };
+
+    this.init();
+  }
+
+  /**
+   * Sets initial values
+   */
+  init() {
+    this.newEnvironmentName = this.environmentName;
+    this.environmentValue = this.workspaceConfig.environments[this.environmentName];
+    this.environmentRecipe = this.parseRecipe(this.environmentValue.recipe.content, this.environmentValue.recipe.contentType);
+  }
+
+  /**
+   * Parses recipe content
+   * @param recipeContent {string}
+   * @param recipeContentType {string}
+   * @returns {object}
+   */
+  parseRecipe(recipeContent, recipeContentType) {
+    let recipe = {};
+    if (/yaml/i.test(recipeContentType)){
```

Comment:
this will work only if type is "compose", in other cases (ex. type: "dockerimage") need to return raw recipeContent